src/parquet/column/reader.h

```diff
@@ -95,6 +100,11 @@ class ColumnReader {
   // The number of values from the current data page that have been decoded
   // into memory
   int num_decoded_values_;
+
+  // We need to store the data page for BYTE_ARRAY and FLBA since the corresponding types
+  // store pointers into these pages
+  // The user is responsible for freeing these DataPages by calling ClearDataPages()
+  std::vector<std::shared_ptr<Page> > byte_pages_;
```

Comment:
per the top level comment let's remove this and add a TODO to `plain-encoding.h` around BA/FLBA that the returned objects do not currently own their data (and thus have lifetime tied to the input stream). Fixing this in a clean way that can yield good performance is too far out of scope for this JIRA